### ISSUE TITLE
[Feature Request] Import and Refresh Files from External Direct download links (URLs)

### ISSUE THREAD

**Issue by elyurn** [iamhyc/Overleaf-Workshop#58]
One can import a file in Overleaf with a download link on the internet. 
In the web app, it's possible to refresh the file (that may have changed). Could you add a right-click with this option on files imported from the internet please?

**Comment by iamhyc** [comment]
Hi @elyurn , I am confused with the expression.

Firstly, I don't think there is a "refresh" option on overleaf webpage (if that is the "web app" you said).
Any changes happened on a file should trigger content refresh itself, wherever on the webpage or in vscode extension.

Secondly, what is a "download link", and how do you import files from a "download link"? Could you demonstrate it with screenshots or gifs?

**Comment by elyurn** [comment]
Hi @iamhyc! Sorry if I wasn't clear. Here is the type of files I import in Overleaf: _From External URL_
![image](https://github.com/iamhyc/Overleaf-Workshop/assets/64151451/14b40f56-9ba7-446a-8b30-aca3bdb22f22)

You can find here the documentation about Direct download links: https://www.overleaf.com/learn/how-to/How_to_upload_a_file_using_an_external_URL
You can see in the video for example that we can refresh to re-sync the file. I would like to have the same feature in VScode. For example, a right click on any file imported from external URLs could show an option to refresh the file from its URL.

Basically, it is useful to sync a bib file online and have it available in any project.

Second, I don't think that it is implemented yet in the Overleaf webpage (i.e., "web app"), but I would love to have an option to set an automatic refresh of files imported from external URLs (so it loads again the download link from URL). I could open a new issue if needed for a feature request since it is not the exact same question.

I hope that I made it clear. Please don't hesitate to ask me again.

**Comment by iamhyc** [comment]
@elyurn Thanks for the information! The video example attached in the above link explains a lot!
I have modified the title to make it more clear.

I am using a open-source overleaf version so I am not aware of this feature. Seems very interesting and useful to hold and sync external files outside overleaf.

I will firstly investigate if it is also supported by open-source overleaf version. Even it is not, we can arrange the feature as part of the *extended API*. Will let you know if there is any progress.

**Comment by iamhyc** [comment]
Hi @elyurn , the feature is now workable in a temporary version. Please try the extension package from this auto-generated artifact here: https://github.com/iamhyc/Overleaf-Workshop/actions/runs/7418535033?pr=62

You can now right-click on the explorer view to "Import File", or right-click on a file to refresh it.

![](https://github.com/iamhyc/Overleaf-Workshop/assets/9068301/c7baf6bf-7cd7-47c1-81fd-06e5db49db99)

Due to the restrictions of VS Code API, the linked files currently shows not difference from a normal file, and the original source to the file could not be displayed.

**Comment by elyurn** [comment]
I just tested the feature. It's great, thank you very much @iamhyc and @QianrenLi for your responsiveness!

Concerning the "import file from another project", I think that it would be a good idea to organize the project search bar using their tags. Right now, I have problems with project sharing the same name under different tags: I can't distinguish one project from another in the search bar.

**Comment by iamhyc** [comment]
> I just tested the feature. It's great, thank you very much @iamhyc and @QianrenLi for your responsiveness!
> 
> Concerning the "import file from another project", I think that it would be a good idea to organize the project search bar using their tags. Right now, I have problems with project sharing the same name under different tags: I can't distinguish one project from another in the search bar.

I see the Overleaf web editor does not display the tag as well. I will see if we can improve it.

**Comment by elyurn** [comment]
> I see the Overleaf web editor does not display the tag as well. I will see if we can improve it.

True. I don't use this feature so I didn't see that, but it's clearly also an issue in the web editor.

**Comment by iamhyc** [comment]
![](https://github.com/iamhyc/Overleaf-Workshop/assets/9068301/0a56b219-133c-411b-b0f8-ce37e2ce722b)

Now we fix this little bug ahead of Overleaf :)

I will merge the PR now and close this issue. This feature will be available in the official release of `v0.7.1`.
